plugins/react.js

```diff
@@ -5,12 +5,23 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
+// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+const {
+  default: jsxDevTransform
+} = require('@babel/plugin-transform-react-jsx-development');
+
 module.exports = {
   plugins: ['react', 'react-hooks', 'jsx-a11y'],
   extends: ['plugin:jsx-a11y/recommended'],
   parserOptions: {
     ecmaFeatures: {
       jsx: true
+    },
+    requireConfigFile: false,
+    babelOptions: {
+      plugins: [
+        jsxDevTransform
+      ]
```

Comment:
No resorting to `preset: '@babel/react-preset`. We only needed `@babel/plugin-transform-react-jsx-development` to make this work for `svg-icons`.

Reply:
Why not just use the preset, which doesn't require the `eslint-disable` clauses (inside the eslint-config 😬)?

Also, this should be added to the index where `jsx: true`.

Reply:
After a convo offline, we will go with `@babel/preset-react` as the simplest option.

Reply:
> Also, this should be added to the index where `jsx: true`.

Also per offline convo, might be best to consolidate all the react/jsx stuff right here. In which case, we handle the peers a bit differently (aka, they aren't spelled out in the package.json)